### PR TITLE
raise exception when bootstrap options doesn't exist in machine options

### PR DESCRIPTION
**Pull request by chilicheech**
Currently, if the machine_options doesn't have a bootstrap_options key we get the following error: `NoMethodError: undefined method `[]' for nil:NilClass`. This PR makes it so an exception is raised instead with a better error/debugging message.

Oh, BTW, I updated the tests to make them pass. Maybe we can re-enable the tests in the travis ci build later.

### CODE REVIEW

**Comment by bdangit** [comment]
:+1: 

hey @tyler-ball can you help get this reviewed so we can use it?

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 sudo: false
 language: ruby
 rvm:
-- 2.1.2
+- 2.2.3
```
Comment (tyler-ball) [inline .travis.yml]: Will you put `2.1.6` in here in addition to `2.2.3` please?

File: spec/unit/providers/aws/credentials_spec.rb [hunk 2]
```diff
@@ -1,45 +1,47 @@
 require 'chef/provisioning/fog_driver/providers/aws/credentials'
 
 describe Chef::Provisioning::FogDriver::Providers::AWS::Credentials do
-  describe "#load_ini" do
+  subject { Chef::Provisioning::FogDriver::Providers::AWS::Credentials.new }
```
Comment (tyler-ball) [inline spec/unit/providers/aws/credentials_spec.rb]: I actually prefer something like `let(:credentials) { Chef::Provisioning::FogDriver::Providers::AWS::Credentials.new }` as opposed to the `subject` magic.  I think it is more clear.  Can you make that change?

File: spec/support/aws/config-file.csv [hunk 3]
```diff
@@ -1,2 +1,2 @@
 User Name,Access Key Id,Secret Access Key
-test,12345,abcde
+test,67890,fghij
```
Comment (tyler-ball) [inline spec/support/aws/config-file.csv]: What is the reason for this change?

File: spec/unit/providers/aws/credentials_spec.rb (hunk 2 above)
Comment (chilicheech) [inline spec/unit/providers/aws/credentials_spec.rb]: I used subject because it's consistent with the other specs in this project. I can change to let if you prefer. Let me know.

File: spec/support/aws/config-file.csv (hunk 3 above)
Comment (chilicheech) [inline spec/support/aws/config-file.csv]: I changed this just to assert that the data that is read in is different from the data that resides in the `ini-file.ini`. Basically to give a bit more confidence that the test is asserting the right thing.

File: .travis.yml (hunk 1 above)
Comment (chilicheech) [inline .travis.yml]: Done
